src/EditorFeatures/Core/Extensibility/NavigationBar/WrappedNavigationBarItem.cs

```diff
@@ -35,16 +35,17 @@ internal WrappedNavigationBarItem(ITextVersion textVersion, RoslynNavigationBarI
 
         private static ImmutableArray<TextSpan> GetSpans(RoslynNavigationBarItem underlyingItem)
         {
-            return underlyingItem is RoslynNavigationBarItem.SymbolItem symbolItem && symbolItem.Location.InDocumentInfo != null
-                ? symbolItem.Location.InDocumentInfo.Value.spans
-                : ImmutableArray<TextSpan>.Empty;
-        }
-
-        private static TextSpan? GetNavigationSpan(RoslynNavigationBarItem underlyingItem)
-        {
-            return underlyingItem is RoslynNavigationBarItem.SymbolItem symbolItem && symbolItem.Location.InDocumentInfo != null
-                ? symbolItem.Location.InDocumentInfo.Value.navigationSpan
-                : null;
+            return underlyingItem switch
+            {
+                // For a regular symbol we want to select it if the user puts their caret in any of the spans of it in this file.
+                RoslynNavigationBarItem.SymbolItem { Location.InDocumentInfo: { } symbolInfo } => symbolInfo.spans,
+
+                // An actionless item represents something that exists just to show a child-list, but should otherwise
+                // not navigate or cause anything to be generated.  However, we still want to automatically select it whenever
+                // the user puts their caret in any of the spans of its child items in this file.
```

Comment:
worth adding an example (maybe vb event and handlers?) that are visually shown in the navbar so it's easy to understand this comment.

Reply:
sure.